src/main/java/Game.java

```diff
@@ -162,4 +172,134 @@ private Position getPositionFromUser() {
         }
     }
 
+    @NotNull
+    private Player getTheWinner() {
+        for (Map.Entry<Position, Cell> entry : board) {
+            Position currentPosition = entry.getKey();
+            checkFreedomLine(currentPosition);
+        }
+        if(blackScore.getNumberOfFreedomLines() > whiteScore.getNumberOfFreedomLines()) {
+            return blackPlayer;
+        }
+        return whitePlayer;
+    }
+
+
+    private void checkFreedomLine(@NotNull Position position) {
+        Stone.Color color = board.getStone(position).getColor();
+        if (countHorizontally(position, 1) == 4) {
+            if(color == Stone.Color.BLACK) {
+                blackScore.incrementNumberOfFreedomLines();
+            } else {
+                whiteScore.incrementNumberOfFreedomLines();
+            }
+        }
+        if (countVertically(position, 1) == 4) {
+            if(color == Stone.Color.BLACK) {
+                blackScore.incrementNumberOfFreedomLines();
+            } else {
+                whiteScore.incrementNumberOfFreedomLines();
+            }
+        }
+        if (countDiagonallyRight(position, 1) == 4) {
+            if(color == Stone.Color.BLACK) {
+                blackScore.incrementNumberOfFreedomLines();
+            } else {
+                whiteScore.incrementNumberOfFreedomLines();
+            }
+        }
+        if (countDiagonallyLeft(position, 1) == 4) {
```

Comment:
What about multiples of 4?

Reply:
??

Reply:
For example, if a player has all cells occupied from A1 up  to A8

Reply:
There must be exactly 4 stones